### PR TITLE
Log of blocking a user

### LINKED ISSUE #200: [QA] brute_force_protection does not log anything
Seen with brute_force_protection 1.2.0 (and also 1.1.0)

* enable app.
* try login a user 4 times with wrong password.
* the user gets blocked. OK
* inspect the owncloud log
```
{"reqId":"ZBGoK7qJUG845N2jOmyzTAAAAAQ","level":2,"time":"2023-03-15T11:12:43+00:00","remoteAddr":"46.114.209.20","user":"--","app":"core","method":"POST","url":"\/login?user=dave","message":"Login failed: 'dave' (Remote IP: '46.114.209.20')"}
{"reqId":"ZBGoPh1eqxX9TLjz1oNVMwAAAAA","level":2,"time":"2023-03-15T11:13:02+00:00","remoteAddr":"46.114.209.20","user":"--","app":"core","method":"POST","url":"\/login?user=dave","message":"Login failed: 'dave' (Remote IP: '46.114.209.20')"}
{"reqId":"ZBGoQrCK0@iLOse6z7CnUwAAAAg","level":2,"time":"2023-03-15T11:13:07+00:00","remoteAddr":"46.114.209.20","user":"--","app":"core","method":"POST","url":"\/login?user=dave","message":"Login failed: 'dave' (Remote IP: '46.114.209.20')"}
{"reqId":"ZBGoph1eqxX9TLjz1oNVNwAAAAA","level":2,"time":"2023-03-15T11:14:46+00:00","remoteAddr":"46.114.209.20","user":"--","app":"core","method":"POST","url":"\/login?user=dave","message":"Login failed: 'dave' (Remote IP: '46.114.209.20')"}
{"reqId":"hEmaOfdKA1K3p60DlpQO","level":1,"time":"2023-03-15T11:15:02+00:00","remoteAddr":"","user":"--","app":"cron","method":"--","url":"--","message":"Invalidating tokens older than 2023-03-15T10:55:02+00:00"}
```
* nothing in the log indicates that the user was blocked. BAD

Expected behaviour:
* whenever the user sees 'To many failed attempts. Try again in 5 Minutes' this should also appear in the owncloud log.

### PR DESCRIPTION
**Pull request by prsnbrg**
Fix #200 

Please test and modify the log message if necessary.

The initial log message of a failed login is generated by: [https://github.com/owncloud/core/blob/master/lib/private/User/Manager.php#L272](https://github.com/owncloud/core/blob/master/lib/private/User/Manager.php#L272)

### CODE REVIEW

**Comment by phil-davis** [comment]
Looks good. I tried to login as "harry" 4 times using a wrong password. After the 4th attempt I  got the "Too many failed login attempts" message in the log file. I tried to login again s "harry" with the correct password, and it still gives the "Too many failed login attempts" message - good. I logged in as a different user and that works - good.
```
$ cat data/owncloud.log | grep "Too many"
{"reqId":"PKsJqVThKn3W60LgUTX3","level":2,"time":"2023-05-16T11:17:01+00:00","remoteAddr":"192.168.0.13","user":"--","app":"brute_force_protection","method":"POST","url":"\/index.php\/login?user=harry","message":"User 'harry' is blocked for  5 minutes. Too many failed login attempts"}
```

File: lib/Throttle.php [hunk 1]
```diff
@@ -88,6 +95,7 @@ public function applyBruteForcePolicyForLogin($uid, $ip) {
 		$thresholdTime = $lastAttempt - $this->config->getBruteForceProtectionTimeThreshold();
 		if ($this->loginAttemptMapper->getFailedLoginCountForUidIpCombination($uid, $ip, $thresholdTime) >=
 			$this->config->getBruteForceProtectionFailTolerance()) {
+			$this->logger->warning('User \''. $uid .'\' is blocked for '. $this->parseBanPeriodForHumans($banPeriod) .'. Too many failed login attempts', ['app' => 'brute_force_protection']);
```
Comment (phil-davis) [inline lib/Throttle.php:98]:
```suggestion
			$this->logger->warning('User \''. $uid .'\' is blocked for'. $this->parseBanPeriodForHumans($banPeriod) .'. Too many failed login attempts', ['app' => 'brute_force_protection']);
```

`parseBanPeriodForHumans` seems to put a leading space in the returned string, so we do not need a space after "for".

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@jvillafanez @jnweiger please review.

File: lib/Throttle.php [hunk 2]
```diff
@@ -88,6 +95,7 @@ public function applyBruteForcePolicyForLogin($uid, $ip) {
 		$thresholdTime = $lastAttempt - $this->config->getBruteForceProtectionTimeThreshold();
 		if ($this->loginAttemptMapper->getFailedLoginCountForUidIpCombination($uid, $ip, $thresholdTime) >=
 			$this->config->getBruteForceProtectionFailTolerance()) {
+			$this->logger->warning('User \''. $uid .'\' is blocked for'. $this->parseBanPeriodForHumans($banPeriod) .'. Too many failed login attempts', ['app' => 'brute_force_protection']);
```
Comment (jvillafanez) [inline lib/Throttle.php:98]: string interpolation seems better:
```
$humanBanPeriod = $this->parseBanPeriodForHumans($banPeriod);
$this->logger->warning("User '{$uid}' is blocked for {$humanBanPeriod}. Too many failed login attempts", ['app' => 'brute_force_protection']);
```

**Review by jvillafanez — COMMENTED** [review]
Just a small fix on the code style. Code looks good

File: lib/Throttle.php (hunk 2 above)
Comment (phil-davis) [reply lib/Throttle.php]: done

**Review by phil-davis — APPROVED** [review]

**Pull request merged**